storage/store_test.go

```diff
@@ -1106,3 +1107,40 @@ func TestRaftNodeID(t *testing.T) {
 		}()
 	}
 }
+
+// BenchmarkLookupAndAddRange adds b.N ranges with randomly generated
+// start/end keys. Before adding a range, looks up if there is any
+// range that has the same key.
+func BenchmarkLookupAndAddRanges(b *testing.B) {
+	defer leaktest.AfterTest(b)
+	store, _, stopper := createTestStore(b)
+	defer stopper.Stop()
+
+	// Remove range 1.
+	rng1, err := store.GetRange(1)
+	if err != nil {
+		b.Fatal(err)
+	}
+	if err := store.RemoveRange(rng1); err != nil {
+		b.Fatal(err)
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		raftID := int64(i)
+		for {
+			start := rand.Int63()
+			startKey := proto.Key(fmt.Sprintf("a%d", start))
```

Comment:
This will generate overlapping ranges, since some values may fall lexicographically between others. I'm not sure what exactly will happen in that case, but it would be better to pad the numbers out to a fixed width so that `start` and `start+1` are lexicographically adjacent.